output.c

```diff
@@ -194,6 +195,18 @@ EXPORT void write_metainfo(FILE *f, struct metafile *m, unsigned char *hash_stri
 		fprintf(f, "6:source%lu:%s",
 			(unsigned long) strlen(m->source), m->source);
 
+	if (m->cross_seed) {
+		unsigned char rand_byte;
+		int value_length = 16;
+		srand(time(NULL));
```

Comment:
I think this would be better in `main()` (or even `init()`), since ideally this a function that is usually called only once throughout the whole program in an initialization step.

Reply:
Good point. I moved it into `main()` right before `init()` is called. Also, I switched to using `srandom()`/`random()` seeded with `clock_gettime()` for better randomness.